osafw-app/App_Code/fw/ConvUtils.cs

```diff
@@ -236,4 +241,199 @@ private static string _replace_specials(string html_data)
         html_data = html_data.Replace(((char)(174)).ToString(), "<sup><small>R</small></sup>");
         return html_data;
     }
+
+    public static Dictionary<string, int> GetMaxCharacterWidth(ArrayList rows, List<string> headers)
+    {
+        var maxColWidth = new Dictionary<string, int>();
+
+        foreach (string header in headers)
+        {
+            var cell = header;
+            var cellValue = header;
+
+            maxColWidth.Add(cell, cell.Length < 10 ? 10 : cell.Length);
+        }
+
+        foreach (Hashtable cells in rows)
+        {
+            foreach (string cell in cells.Keys)
+            {
+                var cellValue = Utils.toStr(cells[cell]);
+                var cellTextLength = cellValue.Length;
+
+                if (!maxColWidth.ContainsKey(cell))
+                {
+                    maxColWidth.Add(cell, cell.Length == 0 ? 50 : cell.Length);
+                }
+
+                if (cellTextLength > maxColWidth[cell])
+                {
+                    maxColWidth[cell] = cellTextLength;
+                }
+            }
+        }
+
+        return maxColWidth;
+    }
+
+    public static DocumentFormat.OpenXml.Spreadsheet.Columns AutoSizeCells(ArrayList rows, List<string> headers)
+    {
+        var maxColWidth = GetMaxCharacterWidth(rows, headers);
+        var columns = new DocumentFormat.OpenXml.Spreadsheet.Columns();
+        double maxWidth = 10;
+
+        UInt32Value iter = 1;
+        foreach (string item in headers)
+        {
+            var val = maxColWidth[item];
+            var width = ((val * maxWidth + 5) / maxWidth * 256) / 256;
+            DocumentFormat.OpenXml.Spreadsheet.Column col = new DocumentFormat.OpenXml.Spreadsheet.Column();
+            col.BestFit = true;
+            col.Min = iter;
+            col.Max = iter;
+            col.CustomWidth = true;
+            col.Width = (double)width;
+            columns.Append(col);
+            iter += 1;
+        }
+        return columns;
+    }
+
+    public static Stylesheet GetStylesheet()
+    {
+        var _Fonts = new DocumentFormat.OpenXml.Spreadsheet.Fonts();
```

Comment:
also, you already `using DocumentFormat.OpenXml.Spreadsheet`, so can simplify everything below?